Packs/FireEyeHX/Integrations/FireEyeHXEventCollector/FireEyeHXEventCollector.py

```diff
@@ -0,0 +1,241 @@
+        ),
+        raw_response=events,
+        outputs=events,
+        outputs_prefix='FireEyeHX.Event'
+    )
+
+
+def fetch_events(
+    client: Client, max_fetch: str, first_fetch: str, resolution: str = None, min_id: str = None,
+        should_push_events: bool = True
+) -> list:
+    """
+    Fetches events from fireeye.
+    """
+
+    # incase we want to retrieve all the resolution type, no need for this parameter.
+    if resolution and resolution.lower() == 'all':
+        resolution = None
+
+    to_date = (datetime.now() + timedelta(days=1)).strftime(DATE_FORMAT)
+    filter_query = {'operator': 'between', 'arg': [first_fetch, to_date], 'field': 'reported_at'}
+
+    response = client.get_events_request(max_fetch, filter_query=json.dumps(filter_query), resolution=resolution,
+                                         min_id=min_id)
+
+    fetched_events = response.get('data', {}).get('entries', [])
+    demisto.info(f'fetched events length: ({len(fetched_events)})')
+
+    populate_modeling_rule_fields(fetched_events)
+
+    if not should_push_events:
+        return fetched_events
+
+    if fetched_events:
+        last_alert: dict = fetched_events[-1]
+        demisto.setLastRun({'last_alert_id': str(last_alert.get('_id')),
+                            'last_alert_time': last_alert.get('event_at')})
+    try:
+        demisto.info(f'sending the following amount of events into XSIAM: {len(fetched_events)}')
+        send_events_to_xsiam(
+            events=fetched_events,
+            vendor=VENDOR,
+            product=PRODUCT
+        )
+    except Exception as e:
+        demisto.info(f'got error when trying to send events to XSIAM: [{e}]')
+
+    return fetched_events
+
+
+def populate_modeling_rule_fields(events: list):
+    for event in events:
+        try:
+            event_date = arg_to_datetime(event.get('event_at'))
+            if event_date:
+                event['_time'] = timestamp_to_datestring(event_date.timestamp() * 1000)
+        except TypeError:
+            # modeling rule will default on ingestion time if _time is missing
+            pass
+
+
+def main() -> None:  # pragma: no cover
+    params = demisto.params()
+    args = demisto.args()
+
+    username = params.get('credentials', {}).get('identifier', '')
+    password = params.get('credentials', {}).get('password', '')
+    base_url = params.get('url', '').rstrip('/')
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    should_push_events = argToBoolean(args.get("should_push_events", False))
+    resolution = params.get('resolution')
+    max_fetch = args.get('limit') or params.get('max_fetch')
+
+    last_run = demisto.getLastRun()
+
+    if last_run.get('last_alert_time'):
+        last_fetch = last_run.get('last_alert_time')
+    else:
+        first_fetch = params.get("first_fetch") or "3 days"
+        last_fetch = arg_to_datetime(first_fetch).strftime(DATE_FORMAT)  # type: ignore[union-attr]
```

Comment:
why not to insert all this to the `fetch_events` function?

Reply:
because `get_events_command` also using `fetch_events` inside
so when we call `fetch-events` command **first_fetch** param is from the `demisto.getLastRun()` 
and when we call `fireeye-hx-get-events` command **first_fetch** param is from the `since` argument.
in my opinion is better to call `demisto.getLastRun()` in the main method instead of check it inside `fetch-events`